app/src/main/java/com/example/tasky/feature_login/domain/model/AuthenticationViewState.kt

```diff
@@ -1,7 +1,9 @@
 package com.example.tasky.feature_login.domain.model
```

Comment:
is this coupling fine? domain and data

Reply:
No, the direction is like this:
presentation -> domain <- data

The goal with this is to let changes in frequently changing layers (data and presentation) have no impact on less frequently changing layers (domain). If domain depends on data however, a change in your data layer could also cause a change in your domain layer which we want to prevent.

In your case, this class is misplaced in domain however. As the name `AuthenticationViewState` already indirectly says, this is related to something presentational. The only purpose is to use this to show certain data to the user. Any UI, ViewModels, UI state, UI events/actions belong in the presentation layer.

Reply:
this is hard concept for me to grasp.. I'm rechecking all files and making sure they are in the correct location. Will resolve this in next PR.